python/housinginsights/sources/wmata_distcalc.py

```diff
@@ -0,0 +1,234 @@
+import sys
+import csv
+import json
+import requests
+import os
+import time
+#import psycopg2
+import time
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__),os.pardir)))
+from housinginsights.sources.base import BaseApiConn
+import housinginsights.tools.dbtools as dbtools
+import sqlalchemy
+#sys.path.insert(0, '../tools/dbtools.py')
+#import tools.dbtools
+
+class WmataApiConn(BaseApiConn):
+
+    def __init__(self):
+
+        secretsFileName = "../secrets.json"
+
+        now = time.strftime("%Y%m%d")
+        outputDir = "../../../data/raw/wmata/" + now
+
+        if not os.path.exists(outputDir):
+            os.makedirs(outputDir)
+
+        self.distOutputFileName = outputDir + "/dist.csv"
+        self.infoOutputFileName = outputDir + "/wmatainfo.csv"
+
+        print("Will write WMATA_DIST table to {}".format(self.distOutputFileName))
+        print("Will write WMATA_INFO table to {}".format(self.infoOutputFileName))
+
+        #pull API keys
+        self.api_keys = json.loads(open(secretsFileName).read())
+        self.wmata_api_key = self.api_keys['wmata']['api_key']
+        self.mapbox_api_key = {'access_token':self.api_keys['mapbox']['public-token']}
+
+        #write out the wmata info csv
+        self.infoOutputFile = open(self.infoOutputFileName, 'wt')
+        self.infoCsvWriter = csv.writer(self.infoOutputFile)
+        self.infoHeaders = ('code_or_id','type','name','lat','lon','lines','stop_id_or_station_code')
+        self.infoCsvWriter.writerow(('code_or_id','type','name','lat','lon','lines','stop_id_or_station_code'))
+
+        #write out hte dist.csv file
+        self.distOutputFile = open(self.distOutputFileName, 'wt')
+        self.distCsvWriter = csv.writer(self.distOutputFile)
+        self.distHeader = ('Nlihc_id','type','stop_id_or_station_code','dist_in_miles')
+        self.distCsvWriter.writerow(('Nlihc_id','type','stop_id_or_station_code','dist_in_miles'))
+        
+        #Creat railStation object
```

Comment:
I was talking about lines 52 and 53 -self.railStations and self.writeBusInfo. Since each of these methods actually query the API and then write the information to CSV's, we shouldn't have them run during object initialization; instead, they should be user driven and callable if desired.